static/js/muting_ui.js

```diff
@@ -113,7 +112,7 @@ exports.mute = function (stream_id, topic) {
         title_text: i18n.t("Topic muted"),
         undo_button_text: i18n.t("Unmute"),
     });
-    exports.set_up_muted_topics_ui(muting.get_muted_topics());
+    // exports.set_up_muted_topics_ui(muting.get_muted_topics());
```

Comment:
With this line I cannot mute a topic unless the settings page has been opened first.
`const muted_topics_table = $("#muted_topics_table").expectOne();` this is the line causing the error (attached image below).

I think this is happening because the `#muted-topic-settings` html element hasn't gotten loaded yet, so jquery can't select it.
idk, if we require this line as all topic mutes are done from the homepage and after commenting it out, the above bug seems to be resolved.
And it doesn't seem this line is used for real time sync, as on opening 2 tabs, if a topic gets muted and if the other tabs is on the `#settings/muted-topics`page, the table gets updated only after reloading.

im not sure how to proceed with this, could i get some help?
![Screenshot from 2020-02-06 18-49-52](https://user-images.githubusercontent.com/40122794/73948019-67f24a80-491e-11ea-9577-6515e1dea7bb.png)

Reply:
@pragatiagrawal31 @aero31aero @hackerkid does one of you have time to help with this?

Reply:
I don't know why it was originally written this way but we do not need to call the `set_up_muted_topics_ui` function here; we already call it when we open the settings UI (see `static/settings_muted.js`). We should remove these two calls:

```diff
diff --git static/js/muting_ui.js static/js/muting_ui.js
index b473303624..bb8a51480f 100644
--- static/js/muting_ui.js
+++ static/js/muting_ui.js
@@ -112,7 +112,6 @@ exports.mute = function (stream_id, topic) {
         title_text: i18n.t("Topic muted"),
         undo_button_text: i18n.t("Unmute"),
     });
-    // exports.set_up_muted_topics_ui(muting.get_muted_topics());
 };
 
 exports.unmute = function (stream_id, topic) {
@@ -124,7 +123,6 @@ exports.unmute = function (stream_id, topic) {
     unread_ui.update_unread_counts();
     exports.rerender();
     exports.persist_unmute(stream_id, topic);
-    exports.set_up_muted_topics_ui(muting.get_muted_topics());
     feedback_widget.dismiss();
 };
```

Reply:
I wonder if there are other similar bugs that have gone by undetected because we do not use `expectOne` or similar logic everywhere.